cli/commands.go

```diff
@@ -437,11 +437,46 @@ func ImportImage(c *cli.Context) error {
 
 // AddNode adds a node to an existing cluster
 func AddNode(c *cli.Context) error {
+
+	/*
+	 * (0) Check flags
+	 */
+
 	clusterName := c.String("name")
-	nodeRole := c.String("role")
 	nodeCount := c.Int("count")
 
-	// check if cluster (i.e. the server) exists
+	/* (0.1)
+	 * --role
+	 * Role of the node that has to be created.
+	 * One of (server|master), (agent|worker)
+	 */
+	nodeRole := c.String("role")
+	if nodeRole == "worker" {
+		nodeRole = "agent"
+	}
+	if nodeRole == "master" {
+		nodeRole = "server"
+	}
+
+	// TODO: support this
+	if nodeRole == "server" {
```

Comment:
The error checking can be improved:

azhou@Andys-MacBook-Pro:~/projs/test/k3d$ bin/k3d add-node -n review --role test
2019/09/25 14:30:35 INFO: Adding 1 test-nodes to k3d cluster review...
2019/09/25 14:30:35 INFO: Created test-node with ID


azhou@Andys-MacBook-Pro:~/projs/test/k3d$ bin/k3d add-node -n review --role worker
2019/09/25 14:31:33 INFO: Adding 1 agent-nodes to k3d cluster review...
2019/09/25 14:31:33 INFO: Created agent-node with ID c7d3750a0bfd45b7b51abf7e015db85eaea3e070c9b4acc4a5dc63e13b91484e

as far as I can tell, using the wrong role name does not add any node.

Reply:
Oh... whoops :D
Fixed (not pretty) in 1fabe11